src/core/xref-db.js

```diff
@@ -0,0 +1,79 @@
+// @ts-check
+import { IDBKeyVal } from "./utils.js";
+
+/**
+ * @typedef {import('core/xref').RequestEntry} RequestEntry
+ * @typedef {import('core/xref').Response} Response
+ * @typedef {import('core/xref').SearchResultEntry} SearchResultEntry
+ */
+
+const CACHE_MAX_AGE = 86400000; // 24 hours
+
+/**
+ * Temporary workaround until browsers get import-maps
+ */
+async function importIdb() {
+  try {
+    return await import("idb");
+  } catch {
+    return await import("../../node_modules/idb/build/esm/index.js");
+  }
+}
+
+async function getIDBKeyValForXref() {
+  const { openDB } = await importIdb();
+  const idb = await openDB("xref", 1, {
+    upgrade(db) {
+      db.createObjectStore("xrefs");
+    },
+  });
+  return new IDBKeyVal(idb, "xrefs");
+}
+
+/**
+ * @param {RequestEntry[]} uniqueQueryKeys
+ * @returns {Promise<Map<string, SearchResultEntry[]>>}
+ */
+export async function resolveXrefCache(uniqueQueryKeys) {
+  const cache = await getIDBKeyValForXref();
+  return await resolveFromCache(uniqueQueryKeys, cache);
+}
+
+/**
+ * @param {RequestEntry[]} keys
+ * @param {IDBKeyVal} cache
+ * @returns {Promise<Map<string, SearchResultEntry[]>>}
+ */
+async function resolveFromCache(keys, cache) {
+  const cacheTime = await cache.get("__CACHE_TIME__");
```

Comment:
Warning: This can throw if cache in't available. Happens in Firefox private window (See: https://github.com/w3c/respec/issues/2288)

Reply:
Oops, accidentally removed try-catch on callers